tabular/src/autogluon/tabular/models/text_prediction/text_prediction_v1_model.py

```diff
@@ -0,0 +1,270 @@
+                import ag_text_prediction_params, merge_params, get_column_properties, \
+                infer_problem_type, infer_eval_stop_log_metrics
+            from autogluon.text.text_prediction.models.basic_v1 import BertForTextPredictionBasic
+        except ImportError:
+            raise ImportError(AG_TEXT_IMPORT_ERROR)
+
+        # Decide the name of the label column
+        if 'label' in X_train.columns:
+            label_col_id = 0
+            while True:
+                self._label_column_name = 'label{}'.format(label_col_id)
+                if self._label_column_name not in X_train.columns:
+                    break
+                label_col_id += 1
+        else:
+            self._label_column_name = 'label'
+        feature_column_properties = get_column_properties(
+            pd.concat([X_train, X_val]),
+            metadata=None,
+            label_columns=None,
+            provided_column_properties=None
+        )
+        label_column_property = get_column_properties(
+            pd.DataFrame({self._label_column_name: pd.concat([y_train, y_val])}),
+            metadata=None,
+            label_columns=None,
+            provided_column_properties=None
+        )
+        column_properties = collections.OrderedDict(list(feature_column_properties.items()) +
+                                                    list(label_column_property.items()))
+        problem_type, label_shape = infer_problem_type(column_properties=column_properties,
+                                                       label_col_name=self._label_column_name)
+        eval_metric, stopping_metric, log_metrics =\
+            infer_eval_stop_log_metrics(problem_type,
+                                        label_shape=label_shape,
+                                        eval_metric=self.eval_metric,
+                                        stopping_metric=self.stopping_metric)
+        search_space = hyperparameters['models']['BertForTextPredictionBasic']['search_space']
+        self.model = BertForTextPredictionBasic(column_properties=column_properties,
+                                                feature_columns=list(X_train.columns),
+                                                label_columns=[self._label_column_name],
+                                                problem_types=[problem_type],
+                                                label_shapes=[label_shape],
+                                                stopping_metric=stopping_metric,
+                                                log_metrics=log_metrics,
+                                                output_directory=os.path.join(self.path, self.name),
+                                                logger=logger,
+                                                base_config=None,
+                                                search_space=search_space)
+        return column_properties
+
+    def _predict_proba(self, X, **kwargs):
```

Comment:
why not just use the parent? If changes have been made, then you need to add the call to

`X = self.preprocess(X, **kwargs)`

which is currently missing and will cause the model to fail.

Reply:
Removed to directly call the parent.